File: Divide a String Into Groups of Size k.py
class Solution(object):
    def divideString(self, s, k, fill):
        if not s:
            return []
        l=[]
        while 0<len(s):
            if len(s)%k!=0:
                s+=fill
            else:
                break
        x=0
        c=k
        while (len(s)/k)>len(l):
            l.append(s[x:c])
            c=c+k
            x=x+k
        return l

File: Divide a String Into Groups of Size k.java
import java.util.*;
class Solution
{
    public String[] divideString(String s, int k, char fill)
    {
        while (s.length()%k!=0)
        {
            s+=fill;
        }
        List<String> l=new ArrayList<>();
        for(int i=0;i<s.length();i+=k)
        {
            l.add(s.substring(i,i+k));
        }
        return l.toArray(new String[0]);
    }
}
